mobile/library/cc/engine_builder.cc

```diff
@@ -529,7 +555,21 @@ std::unique_ptr<envoy::config::bootstrap::v3::Bootstrap> EngineBuilder::generate
     brotli_filter->set_name("envoy.filters.http.decompressor");
     brotli_filter->mutable_typed_config()->PackFrom(decompressor_config);
   }
-  if (gzip_filter_) {
+#ifdef ENVOY_MOBILE_REQUEST_COMPRESSION
+  if (brotli_compression_filter_) {
```

Comment:
On the C++ engine builder, the `enableXYZCompression` APIs aren't even compiled, so it should be impossible to configure it to be enabled if it's compiled out.

Reply:
As for the builders for other languages, I filed https://github.com/envoyproxy/envoy/issues/25355 to do the same there, we just need a small extra bit of infra since CXX linkopts aren't available to Swift/Kotlin/Java.

Reply:
Turns out that Swift & Objective-C can read the CXX linkopts so I've implemented #25355 for those languages in https://github.com/envoyproxy/envoy/pull/25381.